lib/qa/authorities/getty/tgn.rb

```diff
@@ -0,0 +1,65 @@
+module Qa::Authorities
+  class Getty::TGN < Base
+    include WebServiceBase
+
+    attr_reader :subauthority
+    def initialize(subauthority)
+      @subauthority = subauthority
+    end
+
+    def search q
+      parse_authority_response(json(build_query_url(q)))
+    end
+
+    # get_json is not ideomatic, so we'll make an alias
+    def json(*args)
+      get_json(*args)
+    end
+
+    def build_query_url q
+      query = URI.escape(sparql(untaint(q)))
+      "http://vocab.getty.edu/sparql.json?query=#{query}&_implicit=false&implicit=true&_equivalent=false&_form=%2Fsparql"
+
+    end
+
+    def sparql(q)
+      search = untaint(q)
+      # The full text index matches on fields besides the term, so we filter to ensure the match is in the term.
+      # Retrieve the parentString to help with disambiguation
+      sparql = "SELECT ?s ?name ?par {
+              ?s a skos:Concept; luc:term \"#{search}\";
+                 skos:inScheme <http://vocab.getty.edu/tgn/> ;
+                 gvp:prefLabelGVP [skosxl:literalForm ?name] ;
+                  gvp:parentString ?par .
+              FILTER regex(?name, \"#{search}\", \"i\") .
+            } LIMIT 10"
+    end
+
+    def untaint(q)
+      q.gsub(/[^\w\s-]/, '')
+    end
+
+    def find id
+      json(find_url(id))
+    end
+
+    def find_url id
+      "http://vocab.getty.edu/tgn/#{id}.json"
+    end
+
+    def request_options
+      { accept: 'application/sparql-results+json'}
+    end
+
+    private
+
+    # Reformats the data received from the service
+    # Adds the parentString to the location name to enable disambiguation
+    def parse_authority_response(response)
+      response['results']['bindings'].map do |result|
+        { 'id' => result['s']['value'], 'label' => result['name']['value']  + ' (' + result['par']['value'] + ')' }
```

Comment:
Would it make sense to replace the last two values of the parent term? "World" doesn't seem useful and the continent value likely isn't needed. This might make it easier for a user to quickly browse what results they are getting. Something like result['par']['value'].gsub(/\,[^\,]+\,[^\,]+$/, '')?